Elections_scraper/election_sc.py

```diff
@@ -0,0 +1,158 @@
+    infos = []
+    elections = []
+    for url in URLS:
+        answer = get_answer_2(url)
+        parsed = pull_data_2(answer)
+        header_div = parsed.find_all('table', id='ps311_t1', class_='table')
+        election_div = parsed.find_all('div', class_='t2_470')
+        rows = find_row_2(election_div)
+        election = [election_info(row) for row in rows]
+        info = county_info(header_div)
+        infos.append(info)
+        elections.append(election)
+        save_csv_2(infos,elections)
+
+
+def get_answer_2(url):
+    return requests.get(url)
+
+def pull_data_2(ans):
+    return bs(ans.text, "html.parser")
+
+def find_row_2(tables):
+    return [
+        row
+        for table in tables
+        for row in table.find_all('tr')[2:]
+    ]
+
+def county_info(parsed):
+    for contain in parsed:
+        people = contain.find('td',class_='cislo', headers='sa2').text
+        is_env = contain.find('td',class_='cislo', headers='sa3').text
+        submitted = contain.find('td',class_='cislo', headers='sa5').text
+        return {
+            "Volici v seznamu":people,
+            "Vydané obálky":is_env,
+            "Platné hlasy":submitted
+        }
+def election_info(tr) -> dict:
+    try:
+        party = tr.find_all("td")[1].text
+        votes = tr.find_all('td')[2].text
+        return {
+            "PARTY": party,
+            "VOTES": votes
+        }
+    except AttributeError:
+        print(f'Wrong assigned indexes')
+
+def save_csv_2(data, data2):
+    with open('election_data.csv', 'w',) as csv_file:
+        header = ['Volici v seznamu', 'Vydané obálky', 'Platné hlasy', "Občanská demokratická strana",
+                  'Řád národa - Vlastenecká unie', 'CESTA ODPOVĚDNÉ SPOLEČNOSTI', 'Česká str.sociálně demokrat.',
+                  'Radostné Česko', 'STAROSTOVÉ A NEZÁVISLÍ', 'STAROSTOVÉ A NEZÁVISLÍ', 'Komunistická str.Čech a Moravy',
+                  'Strana zelených', 'ROZUMNÍ-stop migraci,diktát.EU', 'Strana svobodných občanů',
+                  'Blok proti islam.-Obran.domova', 'Občanská demokratická aliance', 'Česká pirátská strana',
+                  'Česká národní fronta', 'Referendum o Evropské unii', 'TOP 09', 'ANO 2011', 'Dobrá volba 2016',
+                  'SPR-Republ.str.Čsl. M.Sládka', 'Křesť.demokr.unie-Čs.str.lid.', 'Česká strana národně sociální',
+                  'REALISTÉ', 'SPORTOVCI', 'Dělnic.str.sociální spravedl.', 'Svob.a př.dem.-T.Okamura (SPD)',
+                  'Strana Práv Občanů']
+        writer = csv.DictWriter(csv_file, fieldnames=header)
+        writer.writeheader()
+
+        for data_row, data2_row in zip(data, data2):
+            row_dict = data_row
+            for party_votes in data2_row:
+                row_dict[party_votes["PARTY"]] = party_votes["VOTES"]
+
+            writer.writerow(row_dict)
+
+def cvs_merger(csv1, csv2):
+    df1 = pd.read_csv(csv1)
+    df2 = pd.read_csv(csv2)
+    result = pd.concat([df1, df2], axis=1, ignore_index=False)
```

Comment:
Pandas can solve the merging but isn't it too much for such a simple operation? Why don't you save the csv with all the data you need right away?

Reply:
Ok, thanks for the review, should I change the code accordingly to your comments  for passing the course?